### PR TITLE
Fix #83 RSPEC-1479 switch with too many cases

### LINKED ISSUE #83: Rule S1479: "switch" statements should not have too many "case" clauses
Implements [RSPEC-1479](https://jira.sonarsource.com/browse/RSPEC-1479)

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/SwitchWithTooManyCaseCheck.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  public SwitchWithTooManyCaseCheck() {
+    super(UastNode.Kind.SWITCH);
+  }
+
+  @Override
+  public void visitNode(UastNode node) {
+    SwitchLike switchLike = SwitchLike.from(node);
+    int casesNb = switchLike.caseNodes().size();
+    if(casesNb > maximumCases) {
+      reportIssue(switchLike.switchKeyword(), String.format("Reduce the number of non-empty switch cases from %d to at most %d.", casesNb, maximumCases));
```
Comment (alban-auzeill) [inline common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/SwitchWithTooManyCaseCheck.java]: The "empty switch case" exception is not in this implementation and it's fine. Remove "non-empty" from the message.

File: common-rule-engine/src/test/files/checks/SwitchWithTooManyCaseCheck/SwitchWithTooManyCaseCheck.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
+class A{
+  void foo() {
+    switch (1) { // Noncompliant {{Reduce the number of non-empty switch cases from 38 to at most 30.}}
+//	^^^^^^
```
Comment (alban-auzeill) [inline common-rule-engine/src/test/files/checks/SwitchWithTooManyCaseCheck/SwitchWithTooManyCaseCheck.java]: Our test framework does not support tab characters. To align issue location, replace tab by spaces

File: common-rule-engine/src/test/files/checks/SwitchWithTooManyCaseCheck/SwitchWithTooManyCaseCheck.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package main
+
+type f struct {
```
Comment (alban-auzeill) [inline common-rule-engine/src/test/files/checks/SwitchWithTooManyCaseCheck/SwitchWithTooManyCaseCheck.go]: This struct seems useless for this test. remove it and the 'x' parameter of the foo function in the 2 go test files.

**Review by alban-auzeill — CHANGES_REQUESTED** [review]

File: common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/SwitchWithTooManyCaseCheck.java (hunk 1 above)
Comment (benzonico) [reply common-rule-engine/src/main/java/org/sonar/commonruleengine/checks/SwitchWithTooManyCaseCheck.java]: interestingly, the emptyness of cases is not tested in SonarJava... Message updated.

**Comment by benzonico** [comment]
@alban-auzeill Review comments applied and fixed.

**Review by alban-auzeill — APPROVED** [review]

**Pull request merged**